### PR TITLE
Implement smoke tests

### PR DESCRIPTION
**Pull request by raphaelmue**
See issue #93

### CODE REVIEW

File: de.dashup.application/src/test/java/de.dashup.test/utils/CucumberHooks.java [hunk 1]
```diff
@@ -8,16 +8,21 @@
...
 public class CucumberHooks {
 
     @Before(order = 1)
-    public void doDatabaseSetup() throws SQLException {
+    public void doDatabaseSetup() throws SQLException, IOException {
         //change param of setHost to true if you need to test on local DB
-        Database.setHost(false);
+        InputStream is = DriverUtil.class.getResourceAsStream("/testing.properties");
+        Properties p = new Properties();
+        p.load(is);
```
Comment (svenleonhard) [inline de.dashup.application/src/test/java/de.dashup.test/utils/CucumberHooks.java]: Please avoid shortening

File: de.dashup.application/src/test/java/de.dashup.test/utils/DriverUtil.java [hunk 2]
```diff
@@ -88,10 +88,10 @@ private static WebDriver createFirefoxDriverInstance() {
     }
 
     public static WebDriver createDriverInstance() throws IOException {
-        InputStream is = DriverUtil.class.getResourceAsStream("./testing.properties");
+        InputStream is = DriverUtil.class.getResourceAsStream("/testing.properties");
         Properties p = new Properties();
         p.load(is);
```
Comment (svenleonhard) [inline de.dashup.application/src/test/java/de.dashup.test/utils/DriverUtil.java]: Same here

File: docs/architectures/testing/test_plan.md [hunk 3]
```diff
@@ -1,57 +1,57 @@
...
 ====================
-### Version 1.0
+### Version 2.0
 
 # Revision history
 
 | Date       | Version | Description                                          | Author           |
 |------------|---------|------------------------------------------------------|------------------|
 | 25/04/2019 | 1.0     | Initial Documentation                                | Raphael Müßeler  |
+| 07/05/2019 | 2.0     | Addition of new test type                            | Raphael Müßeler
```
Comment (fidsusj) [inline docs/architectures/testing/test_plan.md]: add the "|" at the end

File: de.dashup.util/src/main/java/de/dashup/util/string/RandomString.java [hunk 4]
```diff
@@ -9,9 +9,9 @@
 
     private static final String upper = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
 
-    private static final String lower = upper.toLowerCase(Locale.ROOT);
```
Comment (fidsusj) [inline de.dashup.util/src/main/java/de/dashup/util/string/RandomString.java]: Why did you remove the modifiers here?

Comment (raphaelmue) [reply de.dashup.util/src/main/java/de/dashup/util/string/RandomString.java]: For tests

File: de.dashup.model/src/test/java/de/dashup/test/DatabaseUnitTest.java [hunk 5]
```diff
@@ -6,17 +6,15 @@
 import de.dashup.shared.User;
 import de.dashup.util.string.Hash;
 import org.json.JSONArray;
-import org.junit.jupiter.api.Assertions;
-import org.junit.jupiter.api.BeforeAll;
-import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.*;
```
Comment (joshuaschu) [inline de.dashup.model/src/test/java/de/dashup/test/DatabaseUnitTest.java]: Do we really need a * import here?

Comment (raphaelmue) [reply de.dashup.model/src/test/java/de/dashup/test/DatabaseUnitTest.java]: IntelliJ improves all imports in such a way that it is as efficient as possible.
